Make sure the AFS lock is held when a user process in a PAG exits, since
otherwise we may crash.  Pulled from the upstream stable branch, will be
in 1.4.5.

--- openafs.orig/src/afs/LINUX/osi_groups.c
+++ openafs/src/afs/LINUX/osi_groups.c
@@ -594,13 +594,18 @@
 {
     afs_uint32 pag = key->payload.value;
     struct unixuser *pu;
+    int locked = ISAFS_GLOCK();
 
+    if (!locked)
+	AFS_GLOCK();
     pu = afs_FindUser(pag, -1, READ_LOCK);
     if (pu) {
 	pu->ct.EndTimestamp = 0;
 	pu->tokenTime = 0;
 	afs_PutUser(pu, READ_LOCK);
     }
+    if (!locked)
+	AFS_GUNLOCK();
 }
 
 struct key_type key_type_afs_pag =
